redisvl/llmcache/semantic.py

```diff
@@ -74,25 +121,40 @@ def check(
         num_results: int = 1,
         fields: List[str] = ["response"],
     ) -> Optional[List[str]]:
-        """Checks whether the cache contains the specified key."""
+        """Checks whether the cache contains the specified prompt or vector.
+
+        Args:
+            prompt (Optional[str], optional): The prompt to check. Defaults to None.
+            vector (Optional[List[float]], optional): The vector to check. Defaults to None.
+            num_results (int, optional): The number of results to return. Defaults to 1.
+            fields (List[str], optional): The fields to return. Defaults to ["response"].
+
+        Raises:
+            ValueError: If neither prompt nor vector is specified.
+
+        Returns:
+            Optional[List[str]]: The response(s) if the cache contains the prompt or vector.
+        """
         if not prompt and not vector:
             raise ValueError("Either prompt or vector must be specified.")
 
-        query = create_vector_query(
-            return_fields=fields,
+        if not vector:
+            vector = self._provider.embed(prompt)  # type: ignore
+
+        v = VectorQuery(
+            vector=vector,
             vector_field_name="prompt_vector",
-            number_of_results=num_results,
+            return_fields=fields,
+            num_results=num_results,
+            return_score=True,
         )
-        if vector:
-            prompt_vector = array_to_buffer(vector)
-        else:
-            prompt_vector = array_to_buffer(self._provider.embed(prompt))  # type: ignore
-        results = self._index.search(query, query_params={"vector": prompt_vector})
+
+        results = self._index.search(v.query, query_params=v.params)
 
         cache_hits = []
         for doc in results.docs:
             self._refresh_ttl(doc.id)
-            sim = similarity(doc.vector_score)
+            sim = similarity(doc.vector_distance)
             if sim > self.threshold:
```

Comment:
this is the part we could do without if we use range query

Reply:
Could also be added later

Reply:
see previous comment